lewis/core/devices.py

```diff
@@ -112,16 +112,9 @@ def __init__(self, module):
 
         self._device_types = list(get_members(self._module, is_device).values())
 
-        if not self._device_types:
-            # syslog.warning('No devices discovered in module {}.'.format(module.__name__))
-            pass
-        else:
-            pass
-            # syslog.info('Discovered devices in module {}: {}'.format(
-            #    module.__name__, ','.join([x.__name__ for x in self._device_types])))
-
         submodules = get_submodules(self._module)
 
+        self._device_types += list(get_members(submodules.get('device'), is_device).values())
```

Comment:
For consistency it might be nice to do it like for the interfaces and setups (like you suggested in your comment) and look around a bit (in case it's not a single `device.py` but a `device/`-directory with multiple files in it). So basically very similar to how `_discover_interfaces` works, but `_discover_device_types`. In fact the method would be more or less the same (even line 113 would go away). Much of the code could be shared I suppose, it's mostly `is_device` vs `is_adapter`.